### ISSUE TITLE
Action to add Maven Wrapper

### ISSUE THREAD

**Issue by brunoborges** [microsoft/vscode-maven#735]
## Suggestion

Provide an action that adds the Maven Wrapper to an existing Maven project.

**Comment by Eskibear** [comment]
A command "Add Maven Wrapper" should work when there's no wrapper detected in project root. Source of the wrapper can be:

#### Option 1:
If a local `mvn` is found on the machine, we run `mvn -N io.takari:maven:0.7.7:wrapper`. Otherwise we copy [maven wrapper packed in this extension](https://github.com/microsoft/vscode-maven/tree/main/resources/maven-wrapper) to project folder.

Pros: 
We allow to specify maven version ` -Dmaven=<version>`.  (but do they care much about maven version?)

#### Option 2:
Always copy the packed wrapper into project folder.

Pros:
Simple and straightforward, we don't require a local maven. (that's why some developers need wrapper).

**Comment by brunoborges** [comment]
As far as I understand, VS Code Maven Extension already comes with a Maven bundled within. This Maven installation could be used instead, both as a Fallback for when there is no Maven Runtime configured, and as a runtime to execute the Add Maven Wrapper goal.

**Comment by Eskibear** [comment]
> VS Code Maven Extension already comes with a Maven bundled within.

No. To clarify, a Maven **wrapper** (instead of a Maven distribution) is bundled in this extension. That was due to some consideration:
- ~10MB extra size 
- unclear whether there's legal issue

**Comment by brunoborges** [comment]
Got it! It is still possible to use the Maven Wrapper to fall the Maven Wrapper plugin to add the Maven Wrapper to a project :-)

**Comment by Eskibear** [comment]
BTW, here is a quick demo of this feature, triggered from command palatte. I'm thinking about whether we need more entries.

https://user-images.githubusercontent.com/2351748/133648682-0ffac8c3-42d4-4d97-b2c6-d6ad158a9030.mp4

**Comment by brunoborges** [comment]
Option B is certainly faster. I'd only suggest to allow user input (optionally) for which Maven version they want (latest by default).

**Comment by Eskibear** [comment]
Agree to allow user input of Maven version.

**Comment by Eskibear** [comment]
@HvB To unblock yourself, perhaps you can contribute to the project on your own by submitting a pull request. 

You can start from https://github.com/microsoft/vscode-maven/pull/739 which is one of the simple solutions for this.
